nltk/internals.py

```diff
@@ -49,7 +49,10 @@ def convert_regexp_to_noncapturing_parsed(parsed_pattern):
                 raise ValueError('Regular expressions with back-references are not supported: {0}'.format(pattern))
             res_data.append((key, value))
         parsed_pattern.data = res_data
-        parsed_pattern.pattern.groups = 1
+        try:
+            parsed_pattern.pattern.groups = 1
+        except AttributeError:
+            pass
```

Comment:
I agree about the current implementation knowing too much about the internals of sre_parse, and that we should reconsider the need for this. In addition to throwing an exception, we could provide a flag so that anyone who depends on the old behaviour can easily get it back, at least for a while.